### PR TITLE
[prometheus-mysql-exporter] update docker image, readme links & use default labels

### PR DESCRIPTION
**Pull request by monotek**
Signed-off-by: André Bauer <andre.bauer@kiwigrid.com>

#### What this PR does / why we need it:
* update prometheus-mysql-exporter docker image
* use absolute links in readme 
* use k8s recommendet labels
* update cloudsqlproxy docker image 

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes # 

#### Special notes for your reviewer:
related to: https://github.com/prometheus-community/helm-charts/issues/61

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by monotek** [comment]
@juanchimienti could you review please?

File: charts/prometheus-mysql-exporter/README.md
Comment (torstenwalter) [inline charts/prometheus-mysql-exporter/README.md:44]: Another option seems to be to remove selectors https://github.com/jenkinsci/helm-charts/issues/46.

Haven't tested it.

File: charts/prometheus-mysql-exporter/Chart.yaml
Comment (torstenwalter) [inline charts/prometheus-mysql-exporter/Chart.yaml:1]: Do you have a plan to upgrade to v2?

Comment (monotek) [reply charts/prometheus-mysql-exporter/Chart.yaml:1]: I think as soon as helm 2 deprecates?
Chart can still be used with helm 3 until then.

Updating now would mean no helm 2 support.
Do you have any reason to hurry?

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (Xtigyro) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:37]: Isn't it still better not to include the version in the labels?

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (Xtigyro) [inline charts/prometheus-mysql-exporter/templates/_helpers.tpl:48]: Why not call them `matchLabels` - so we follow the current convention? E.g. - https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/templates/_helpers.tpl#L19

**Review by Xtigyro — CHANGES_REQUESTED** [review]

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (monotek) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:48]: Imho the current convention is the helm standard, which is "selectorLabels".
If somebody contributes he probably expects it like in other helmcharts, created via "helm create".

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (monotek) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:37]: Why? Whats the benefit?
It's the default helm 3.3.1 uses with "helm create prometheus-mysql-exporter"

Comment (Xtigyro) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:37]: Clearer Helm upgrade logs maybe?

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (Xtigyro) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:48]: Only Job, Deployment, Replica Set, and Daemon Set support `matchLabels` - therefore, you may see just `selector` for Service (e.g.).

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (monotek) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:37]: I don't get your point?

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (monotek) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:48]: I don't get your point?

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (Xtigyro) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:37]: When you do `helm upgrade` - probably the logs will be cleaner and there will be less shuffling?

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (Xtigyro) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:48]: I mean for those resources `matchLabels` might be the more precise name for such labels.

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (monotek) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:37]: Not sure if less info in the logs is preferable.

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (monotek) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:48]: I think this is opinionated as only proemtheus chart uses it this way.
I'd like to keep the helm default.

Comment (Xtigyro) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:48]: OK - agreed.

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (Xtigyro) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:37]: If you do an upgrade - should you wanna change only the resources you wanna upgrade? If it's how it is - then you will modify resources that are not really affected.

Comment (monotek) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:37]: If you raise the chart version for example the deployment is updated but not the pod. So no restart triggered.

Which resources do have you in mind?

Comment (Xtigyro) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:37]: Yeah, no restart - agreed. Isn't every single resource going to be mentioned in the upgrade logs in this case? That's the only thing that bothers me a little bit because it might be a bit overwhelming though no deal breaker.

Comment (monotek) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:37]: I'll have to check.
I'm used to fluxcd so i don't look at the logs anyway :D

Comment (Xtigyro) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:37]: Bad boy! 🥇

**Review by Xtigyro — APPROVED** [review]

**Comment by monotek** [comment]
Hmmm... Still can't merge. 

@scottrigby 
Imho 1 approver should be enough or not?

**Comment by scottrigby** [comment]
Branch protection requires one code owner approval (matching the CODEOWNERS rules: https://github.com/prometheus-community/helm-charts/blob/main/.github/CODEOWNERS#L17), from someone who is not also the person that opened the PR.

File: charts/prometheus-mysql-exporter/Chart.yaml
Comment (scottrigby) [reply charts/prometheus-mysql-exporter/Chart.yaml:1]: I agree it's best to wait until Nov 13. There is a way to start a chart API v2 version of a chart in the same repo sooner, but let's discuss that separately.

File: charts/prometheus-mysql-exporter/templates/_helpers.tpl
Comment (scottrigby) [reply charts/prometheus-mysql-exporter/templates/_helpers.tpl:37]: This is something I'd like to look into

**Review by scottrigby — CHANGES_REQUESTED** [review]

File: charts/prometheus-mysql-exporter/README.md
Comment (monotek) [reply charts/prometheus-mysql-exporter/README.md:44]: Ups, sorry. Was'nt aware of the order. 
I'll put it back.

File: charts/prometheus-mysql-exporter/README.md
Comment (monotek) [reply charts/prometheus-mysql-exporter/README.md:82]: My vscode markdown linter complained: https://github.com/DavidAnson/markdownlint/blob/v0.20.4/doc/Rules.md#md004

File: charts/prometheus-mysql-exporter/README.md
Comment (monotek) [reply charts/prometheus-mysql-exporter/README.md:92]: My vscode markdown linter complained: https://github.com/DavidAnson/markdownlint/blob/v0.20.4/doc/Rules.md#md004

**Comment by monotek** [comment]
> Branch protection requires one code owner approval (matching the CODEOWNERS rules: https://github.com/prometheus-community/helm-charts/blob/main/.github/CODEOWNERS#L17), from someone who is not also the person that opened the PR.

ok, so i guess an addditional maintainer for the chart would be good....

**Comment by scottrigby** [comment]
Let's give it a little time to hear back from another maintainer of the chart. 

In general for this repo, what what a good wait policy be before a repo admin should step in after not hearing from a maintainer of the PR changed chart - 1 month? 2 weeks? We should decide and document in the PROCESSES file.

**Comment by monotek** [comment]
I think 2 weeks are enough. 1 month would mean that in worst case you could only release 12 versions per year for a chart with more frequent changes.

**Comment by monotek** [comment]
@scottrigby everything should be fixed now. 
could you reveiw again please?

File: charts/prometheus-mysql-exporter/README.md
Comment (scottrigby) [reply charts/prometheus-mysql-exporter/README.md:60]: Thanks this reads much clearer now

**Comment by scottrigby** [comment]
@monotek this lgtm 👍

Yes it has now been 22 days with no response from the chart maintainer. 

We have generally agreed on a 2 week wait time, and @torstenwalter compiled the conversation into the PROCESSES doc in this PR: https://github.com/prometheus-community/helm-charts/pull/147 In the spirit of following good processes, let's wait till that's merged (there's only one more minor change requested) and then this one can be merged just after the new process documentation is in he main branch. Does that make sense? Thanks for your patience on this.

**Comment by monotek** [comment]
@scottrigby 

#147 hast bee merged so I guess we can proceed here too :)

**Comment by monotek** [comment]
@scottrigby 
pretty please :)

**Comment by monotek** [comment]
@torstenwalter could you review please?

**Review by scottrigby — APPROVED** [review]

**Pull request merged**
